boards/silabs/radio_boards/slwrb4180b/slwrb4180b-pinctrl.dtsi

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2024 Capgemini
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <arm/silabs/gpio_gecko.h>
+#include <dt-bindings/pinctrl/gecko-pinctrl.h>
+
+&pinctrl {
+	usart0_default: usart0_default {
+		group1 {
+			psels = <GECKO_PSEL(UART_TX, A, 5)>,   // TX pin
+				<GECKO_PSEL(UART_RX, A, 6)>,   // RX pin
+				<GECKO_LOC(UART, 0)>;          // Location configuration
+		};
+	};
+
+	usart0_spi_default: usart0_spi_default {
+		group1 {
+			psels = <GECKO_PSEL(SPI_SCK, C, 2)>,    // SPI clock pin
+				<GECKO_PSEL(SPI_MISO, C, 1)>,   // SPI MISO pin
+				<GECKO_PSEL(SPI_MOSI, C, 0)>;   // SPI MOSI pin
+		};
+	};
```

Comment:
What is purpose of `usart0_spi_default`? It seems it is orphan.

Reply:
Hi @jerome-pouiller 
Thank you for your feedback. The usart0_spi_default pinctrl is included because, according to the datasheet, USART0 supports both UART and SPI functionalities. The usart0_spi_default configuration defines the necessary pins for SPI mode. Although it may appear orphaned, it provides the required pin mappings for users who need to enable SPI communication through USART0. If a user wants to utilize SPI on USART0, they can reference this pinctrl in their device tree configuration.